sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -35,41 +38,93 @@ namespace Azure { namespace Core { namespace Http {
+    // unique_ptr class wrapping an HINTERNET handle
+    class DeleteHINTERNET {
+    public:
+      void operator()(HINTERNET handle) noexcept
+      {
+        if (handle != nullptr)
+        {
+          WinHttpCloseHandle(handle);
+        }
+      }
+    };
+    /**
+     * @brief unique_ptr class wrapping an HINTERNET handle
+     */
+    using UniqueHINTERNET = std::unique_ptr<void, DeleteHINTERNET>;
+    // unique_ptr class wrapping a PCCERT_CHAIN_CONTEXT
+    struct DeleteHCERTIFICATECHAIN
     {
-      Context const& m_context;
-      Request& m_request;
-      HINTERNET m_connectionHandle;
-      HINTERNET m_requestHandle;
+      void operator()(PCCERT_CHAIN_CONTEXT handle)
+      {
+        // unique_ptr class wrapping an HINTERNET handle
+        {
+          CertFreeCertificateChain(handle);
+        }
+      }
+    };
 
-      HandleManager(Request& request, Context const& context)
-          : m_request(request), m_context(context)
+    /**
+     * @brief unique_ptr class wrapping a "const CERT_CHAIN_CONTEXT" pointer.
+     */
+    using UniqueCCERT_CHAIN_CONTEXT
+        = std::unique_ptr<const CERT_CHAIN_CONTEXT, DeleteHCERTIFICATECHAIN>;
+
+    // unique_ptr class wrapping an HCERTCHAINENGINE handle
+    struct DeleteHCERTCHAINENGINE
+    {
+      void operator()(HCERTCHAINENGINE handle) noexcept
       {
-        m_connectionHandle = NULL;
-        m_requestHandle = NULL;
+        if (handle != nullptr)
+        {
+          CertFreeCertificateChainEngine(handle);
+        }
       }
+    };
+	
+    /**
+     * @brief unique_ptr class wrapping a "HCERTCHAINENGINE" handle.
+     */
+    using UniqueHCERTCHAINENGINE = std::unique_ptr<void, DeleteHCERTCHAINENGINE>;
 
-      ~HandleManager()
+    // unique_ptr class wrapping an HCERTSTORE handle
+    struct DeleteHCERTSTORE
+    {
+    public:
+      void operator()(HCERTSTORE handle) noexcept
       {
-        // Close the handles and set them to null to avoid multiple calls to WinHTTP to close the
-        // handles.
-        if (m_requestHandle)
+        if (handle != nullptr)
         {
-          WinHttpCloseHandle(m_requestHandle);
-          m_requestHandle = NULL;
+          CertCloseStore(handle, 0);
         }
+      }
+    };
+    /**
+     * @brief unique_ptr class wrapping a "HCERTSTORE" handle.
+     */
+    using UniqueHCERTSTORE = std::unique_ptr<void, DeleteHCERTSTORE>;
 
-        if (m_connectionHandle)
+    // unique_ptr class wrapping a PCCERT_CONTEXT
+    struct DeleteCERTCONTEXT
+    {
+    public:
+      void operator()(PCCERT_CONTEXT handle) noexcept
+      {
+        if (handle != nullptr)
         {
-          WinHttpCloseHandle(m_connectionHandle);
-          m_connectionHandle = NULL;
+          CertFreeCertificateContext(handle);
         }
       }
     };
+    /**
+     * @brief unique_ptr class wrapping a "PCCERT_CONTEXT" object.
+     */
+    using UniquePCCERT_CONTEXT = std::unique_ptr<CERT_CONTEXT const, DeleteCERTCONTEXT>;
```

Comment:
The name still does not correspond to our naming guidelines. It has underscore, and has more than 2 capital letters in a row.
The name should be `UniquePccertContext`.

Reply:
I don't know how many times I've said this, but: *We do not rename types from 3rd party libraries*. 

Please stop asking me to rename types that are not a part of the C++ SDK.

Reply:
Sorry, but the naming does not follow our naming guidelines (https://azure.github.io/azure-sdk/cpp_introduction.html#naming), so I have no option but keep the comment active. We can discuss on the C++ meeting on Tuesday.

Here, Rick also seem to agree with me on `UniquePccertContext` naming.

Naming it "PccertContext" is not renaming, the name is the same. We don't name it "PointerToConstCertificateContext" (although I don't see problem with picking a different name if that serves our purposes better. Where's the requirement to not rename comes from?) This is not renaming, we are defining a type within our SDK. `UniqueXxx` is the type that we declare, so it should follow our naming guidelines.

Exactly because we treat `CERTCONTEXT` as a single word, the name should be `DeleteCertcontext` and not `DeleteCertContext`. We follow the same logic with `Url`, `Sha` and `Guid`.

Reply:
The thing Rick was complaining about was a private definition of an OpenSSL type which I added to avoid a breaking change (including OpenSSL headers in a public header is a breaking change). It's completely unrelated to this discussion.

However I implore you to PLEASE TAKE A LOOK AT THE *ACTUAL* PULL REQUEST and stop leaving comments about previous iterations of the pull request.

I'm tired enough of fighting you on this one that I've removed every single UniqueXxx type from the pull request. It significantly complicated the code, but hopefully it will make you happy.

Having said that, I will 100% die on this hill: We *DO NOT* rename 3rd party types. That's because the names 3rd party types  imply certain semantics on those types. If we define convenience wrappers around those 3rd party types, we should preserve the name of the 3rd party types in the convenience wrapper because anything else removes the semantics of those types. 

To me, it's not at all clear that a `Curl` object is the same as a `CURL` object.

PS: I'll also note that we did *NOT* make the same changes when we incorporated the `nlohmann::json` types into the SDK. So the naming guidelines need to be treated as flexible.

Reply:
There is a like from @RickWinter under my unambiguous and single-purposed comment above, so it means agrees with `UniquePccertContext` naming.

I am also not giving up on this hill, let's then resolve during our Tuesday C++ team meeting.